lillisp.go

```diff
@@ -3,15 +3,78 @@ package main
 import (
 	"bufio"
 	"fmt"
+	"log"
 	"os"
+	"strconv"
 	"unicode/utf8"
 )
 
+// Atom ...
+type Atom interface{}
+
+// Pair represents the fundamental "cons-pair" in LISP-like languages
 type Pair struct {
 	car interface{}
 	cdr interface{}
 }
 
+var funcMap = map[string]func(Atom, Atom) Atom{
+	"+": func(a Atom, b Atom) Atom { return toInt(a) + toInt(b) },
+	"-": func(a Atom, b Atom) Atom { return toInt(a) - toInt(b) },
+	"*": func(a Atom, b Atom) Atom { return toInt(a) * toInt(b) },
+	"/": func(a Atom, b Atom) Atom { return toInt(a) / toInt(b) },
+}
+
+func toInt(a Atom) int {
+	if val, ok := a.(int); ok == true {
+		return val
+	} else if val, ok := a.(string); ok == true {
+		if intVal, err := strconv.Atoi(val); err == nil {
+			return intVal
+		}
+	}
+
+	log.Panicf("Could not convert %v to Number\n", a)
+	return 0 // technically will never reach here
```

Comment:
does the compiler complain if you don't have this?

Reply:
Yeah.